eth/polls/eth_subscribe.go

```diff
@@ -88,59 +107,75 @@ func (s EthSubscriptions) AddLogPoll(filter eth.EthFilter, height uint64) (strin
 	}, height), nil
 }
 
-func (s EthSubscriptions) LegacyAddLogPoll(filter string, height uint64) (string, error) {
+func (s *EthSubscriptions) LegacyAddLogPoll(filter string, height uint64) (string, error) {
 	newPoll, err := NewEthLogPoll(filter, s.evmAuxStore, s.blockStore)
 	if err != nil {
 		return "", err
 	}
 	return s.Add(newPoll, height), nil
 }
 
-func (s EthSubscriptions) AddBlockPoll(height uint64) string {
+func (s *EthSubscriptions) AddBlockPoll(height uint64) string {
 	return s.Add(NewEthBlockPoll(height, s.evmAuxStore, s.blockStore), height)
 }
 
-func (s EthSubscriptions) AddTxPoll(height uint64) string {
+func (s *EthSubscriptions) AddTxPoll(height uint64) string {
 	return s.Add(NewEthTxPoll(height, s.evmAuxStore, s.blockStore), height)
 }
 
-func (s EthSubscriptions) AllLogs(
+func (s *EthSubscriptions) AllLogs(
 	state loomchain.ReadOnlyState, id string, readReceipts loomchain.ReadReceiptHandler,
 ) (interface{}, error) {
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
+
 	if poll, ok := s.polls[id]; !ok {
 		return nil, fmt.Errorf("subscription not found")
 	} else {
 		return poll.AllLogs(state, id, readReceipts)
 	}
 }
 
-func (s EthSubscriptions) Poll(
+func (s *EthSubscriptions) Poll(
 	state loomchain.ReadOnlyState, id string, readReceipts loomchain.ReadReceiptHandler,
 ) (interface{}, error) {
-	if poll, ok := s.polls[id]; !ok {
+	s.mutex.Lock()
+	poll, ok := s.polls[id]
+	if !ok {
+		s.mutex.Unlock()
 		return nil, fmt.Errorf("subscription not found")
-	} else {
-		newPoll, result, err := poll.Poll(state, id, readReceipts)
-		s.polls[id] = newPoll
-		s.resetTimestamp(id, uint64(state.Block().Height))
-		return result, err
 	}
+	newPoll, result, err := poll.Poll(state, id, readReceipts)
+	s.polls[id] = newPoll
+	s.mutex.Unlock()
+
+	s.resetTimestamp(id, uint64(state.Block().Height))
```

Comment:
`s.resetTimestamp` is not called anywhere outside this file. I think we can make it a util function and then we can just remove lock mutex inside the function. So we can do `mutex.Lock()` and `defer mutex.Unlock()` in `Poll` and `LegacyPoll` instead to simplify the logic.